clients/ts/signalr/tests/WebSocketTransport.test.ts

```diff
@@ -0,0 +1,203 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+import { ILogger } from "../src/ILogger";
+import { TransferFormat } from "../src/ITransport";
+import { NullLogger } from "../src/Loggers";
+import { WebSocketTransport } from "../src/WebSocketTransport";
+import { VerifyLogger } from "./Common";
+import { TestMessageEvent } from "./TestEventSource";
+import { TestCloseEvent, TestEvent, TestWebSocket } from "./TestWebSocket";
+
+describe("WebSocketTransport", () => {
+    [[TransferFormat.Text, "blob"],
+    [TransferFormat.Binary, "arraybuffer"]]
+        .forEach(([input, expected]) => {
+            it(`sets websocket binarytype based on transferformat: ${TransferFormat[input]}`, async () => {
```

Comment:
The logic isn't to set the binary type to `blob` when sending `Text` and to `arraybuffer` when sending `Binary`. It's actually just that we always want `binaryType` to be `arraybuffer` because that's how we send binary frames. However, we only care about setting it when the `TransferFormat` is `Binary` and we'll never send any binary frames when it's `Text` so we don't bother setting it.

What I'm saying is, I don't think we need to test that the `binaryType` value is `blob` when using `TransferMode.Text`.